src/lua/api-gateway/validation/validator.lua

```diff
@@ -79,6 +80,9 @@ function BaseValidator:setKeyInLocalCache(key, string_value, exptime, dict_name)
 end
 
 function BaseValidator:getRedisUpstream(upstream_name)
+    if self.redisHost then
```

Comment:
I believe you also need a second variable, for the Redis read replica.
I would rather change the static `redis_RO_upstream` and `redis_RW_upstream` variables to class members of the `"api-gateway.validation.validator"` base class have them overwritten when this class is extended by `oauthTokenValidator`.